docs/docs/09-commands-and-tools/02-ide-support.md

```diff
@@ -37,80 +37,156 @@ Include the following to the settings.json in order to enable autocompletion for
+For neovim you can use [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) and install [tree-sitter-templ](https://github.com/vrischmann/tree-sitter-templ) with `:TSInstall templ`.
 
 The configuration for the templ Language Server is included in [lspconfig](https://github.com/neovim/nvim-lspconfig), [mason](https://github.com/williamboman/mason.nvim),
 and [mason-lspconfig](https://github.com/williamboman/mason-lspconfig.nvim).
 
+The `templ` command must be in your system path for the LSP to be able to start. Ensure that you can run it from the command line before continuing.
+
 Installing and configuring the templ LSP is no different to setting up any other Language Server.
 
 ```lua
+local lspconfig = require("lspconfig")
+
 -- Use a loop to conveniently call 'setup' on multiple servers and
 -- map buffer local keybindings when the language server attaches
+
 local servers = { 'gopls', 'ccls', 'cmake', 'tsserver', 'templ' }
 for _, lsp in ipairs(servers) do
-  nvim_lsp[lsp].setup {
+  lspconfig[lsp].setup({
     on_attach = on_attach,
-    flags = {
-      debounce_text_changes = 150,
-    },
-  }
+    capabilities = capabilities,
+  })
 end
 ```
 
-The `templ` command must be in your system path for the LSP to be able to start.
+In Neovim, you can use the `:LspInfo` command to check which Language Servers (if any) are running. If the expected language server has not started, it could be due to the unregistered templ file extension. 
+To resolve this issue, add the following code to your configuration. This is also necessary for other LSPs to "pick up" on .templ files.
 
-In Neovim, you can use the `:LspInfo` command to check which Language Servers (if any) are running. If the expected language server has not started, it could be due to the unregistered templ file extension. To resolve this issue, add the following code to your configuration:
+```lua
+vim.filetype.add({ extension = { templ = "templ" } })
+```
 
+##### Other LSPs within .templ files
+
+These LSPs can be used *in conjunction* with the templ lsp and tree-sitter. Here's how to set them up.
+
+[html-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#html) - First make sure you have it installed `:LspInstall html` or find it on the `:Mason` list. 
 ```lua
--- additional filetypes
-vim.filetype.add({
- extension = {
-  templ = "templ",
- },
+lspconfig.html.setup({
+    on_attach = on_attach,
+    capabilities = capabilities,
+    filetypes = { "html", "templ" },
 })
 ```
 
-### Format on Save
+[htmx-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#htmx) - First make sure you have it installed `:LspInstall htmx` or find it on the `:Mason` list. Note with this LSP, it activates after you type `hx-` in an html attribute, because that's how all htmx attributes are written.
```

Comment:
```suggestion
[htmx-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#htmx) - First make sure you have it installed `:LspInstall htmx` or find it on the `:Mason` list. Note with this LSP, it activates after you type `hx-` in an html attribute, because that's how all htmx attributes are written.

```